### PR TITLE
Sustainability in Agile: How Scrum Roles Can Drive Greener Practices

### PR DESCRIPTION
**Pull request by jcamilleri-scottlogic**
Please add a direct link to your post here:

[https://jcamilleri-scottlogic.github.io/blog-sustainable-agile/2024/10/21/sustainable-agile.html](https://jcamilleri-scottlogic.github.io/blog-sustainable-agile/2024/10/21/sustainable-agile.html)

Have you (please tick each box to show completion):

 - [X] Added your blog post to a single category?
 - [X] Added a brief summary for your post? Summaries should be roughly two sentences in length and give potential readers a good idea of the contents of your post.
 - [X] Checked that the build passes?
 - [X] Checked your spelling (you can use `npm install` followed by `npx mdspell "**/{FILE_NAME}.md" --en-gb -a -n -x -t` if that's your thing)
 - [X] Ensured that your author profile contains a profile image, and a brief description of yourself? (make it more interesting than just your job title!)
 - [X] Optimised any images in your post? They should be less than 100KBytes as a general guide.

Posts are reviewed / approved by your Regional Tech Lead.

### CODE REVIEW

File: _posts/2024-09-03-sustainable-agile.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+title: 'Sustainability in Agile: How Scrum Roles Can Drive Greener Practices'
+date: 2024-09-03 11:00:00 Z
+categories:
+- Sustainability
+tags:
+- Sustainability
```
Comment (dogle-scottlogic) [inline _posts/2024-09-03-sustainable-agile.md]: Could be worth tagging as "Agile" as well

File: _posts/2024-09-03-sustainable-agile.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+At the project level, you just need to focus on the items you want in scope for your team. Using the Upstream Emissions category (U), and Category O of Operational Emissions, you can track the hardware that your team uses to determine the teams own carbon emissions while developing the software. These categories can also be used if you have on premises server hardware that you use for development, testing or in production environments.
+
+Category C is used for cloud environments, software as a service and other IT services that you do not host yourself. If you are unsure what category to put a system into, ask if you pay the electricity bill for that system, or if you pay a third party for access to that system. If you are paying the electricity bill, it is classed as a direct emission (category O). Otherwise, it is indirect (category C).
+
+Your systems also have an impact downstream: 
+- the minimum specification of your software drives hardware sales and therefore old hardware disposal. 
+- software performance and network traffic use the resources of your consumers and the infrastructure used to reach them.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: For some reason these bullets are not rendering correctly. Try an empty line directly before the bullets.

File: _posts/2024-09-03-sustainable-agile.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+## Be Transparent
+
+Once you have mapped out and measured or at least estimated the carbon emissions of your project, find a way to make them available, ideally via a dashboard. There are a few tools that can assist you in this regard:
+
+- [Cloud Carbon Footprint]( https://www.cloudcarbonfootprint.org/): In addition to the dashboards provided by your cloud provider, the CCF tool can be used to get up to date metrics from a variety of cloud providers to create a dashboard for your carbon emissions. Installation ca be a little involved, but once it is set up, it is quite easy to use.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Typo: "ca be" => "can be"

File: _posts/2024-09-03-sustainable-agile.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+- [Scaphandre](https://github.com/hubblo-org/scaphandre): This is an open-source utility for measuring the energy utilisation of processes on a server. It can be configured to output the data to any target such as Prometheus. When no other data is available (such as in an On Prem environment), CPU utilisation, which Scaphandre measures, can be a good proxy for carbon emissions, as higher CPU usage is normally a good indicator of this.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: > good indicator of this

What is "this" here? Do you mean higher CPU usage is normally a good indicator of _higher_ carbon emissions?

File: _posts/2024-09-03-sustainable-agile.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+# How Scrum Roles Can Mitigate Carbon Emissions
+
+Having talked about how we map and measure the carbon emissions of the project, it is down to the team to start reducing carbon emissions. Each role has their part to play.
+
+## If you are the Product Owner
+
+As a product owner, you are already responsible for defining the product’s features, liaising with stakeholders, and prioritising the backlog.
+
+- *Feature Prioritization*: Try to prioritize features that reduce the product's environmental footprint, such as optimizing code efficiency or reducing server load.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Haha, I'm sure that's the first thing on the PO's mind when their ear is being bent by their boss 😆

File: _posts/2024-09-03-sustainable-agile.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+## If you are the Product Owner
+
+As a product owner, you are already responsible for defining the product’s features, liaising with stakeholders, and prioritising the backlog.
+
+- *Feature Prioritization*: Try to prioritize features that reduce the product's environmental footprint, such as optimizing code efficiency or reducing server load. 
+- *Estimation*: Encourage team members to incorporate sustainable considerations into their estimates. Help them to carve out time to craft high quality software – remember, doing so can help reduce your running costs and total cost of ownership by keeping systems well designed and efficient.
+- *Lifecycle Thinking*: Consider the entire product lifecycle, focusing on features that enhance durability and reduce waste.
+
+You can integrate the sustainability metrics discussed above into decision-making, ensuring that carbon reduction is a key factor in planning and prioritization. For example, if you had a user story that would allow a product to use a smaller, or even eliminate a cloud appliance, prioritising it would reflect in the cloud emissions metrics.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Slightly clunky sentence... Maybe this?

> to use a smaller cloud appliance, or even eliminate it entirely

File: _posts/2024-09-03-sustainable-agile.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+As team members, you will be the ones making these smaller changes that will add up in the larger picture, coordinated by your Product Owner. How you do that will depend on your speciality. While we cannot cover all possible specialisms here, we can look at some of the most common.
+
+### As a Software Developer
+
+- *Efficient Coding Practices*: You should aim to write clean, efficient code that reduces computational load and energy consumption.
+- *Sustainable Architecture*: Advocate for and implement architectural designs that minimize resource use, such as serverless architectures or microservices that scale efficiently.
+
+The TCS can be used as a framework for evaluating the impact of code changes. As the TCS grows and improves you will find tips for code and architecture design, and links to current tools that can assist you.
+For some guidance on potentially useful software patterns that promote green coding, take a look at Green Software Foundation’s https://patterns.greensoftware.foundation/. Each of these patterns should be judged on their own merit, with their suitability for your specific product in mind.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Link is plaintext; use this instead:

`[Green Software Foundation patterns](https://patterns.greensoftware.foundation/)`

File: _posts/2024-09-03-sustainable-agile.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+### As a DevOps Engineer
+
+- *Efficient Build Pipelines*: Optimise your build pipelines to run quickly and efficiently, paying attention to what resources are consumed when they run, and how often they run.
+- *Shut Down Unused Processes*: Look for zombie processes that are running, taking up resources, but are not actually being used by anyone. Also keep an eye out for system that are left running when not in use for extended periods. For example, are integration environments only used in the day, but left running all night?
+- *Check the Energy Mix*: Where does your hosting provider get its energy? Is it from renewable sources? If not, is there a viable alternative that can deliver the same feature at a similar or lower cost that runs from renewables? If your software is cloud hosted, could you move to a more sustainable location?
+- *Optimise Regular Jobs*: If there are automated processes that run regularly, check that they are doing so efficiently and are definitely required. It may also be worth check if they can be run at a time or location with a cleaner energy mix.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Typo: "worth check if" => "worth checking if"

File: _posts/2024-09-03-sustainable-agile.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+### As a Tester
+
+- *Develop Carbon Aware Metrics*: Craft tests that can surface carbon emission data and check the feedback over time. If a test shows that the new feature uses a lot of carbon compared to other, similar features, make the developer aware. Similarly, check if a change to an existing feature has improved or worsened the carbon emissions of the system under test.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Any links for how to approach this?

File: _posts/2024-09-03-sustainable-agile.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+Of course, there are a great many other specialised skills that I've not detailed here, but you can use the above to think about how your skills can contribute to the shared goal of improving digital sustainability.
+
+# Collaborative Efforts and Continuous Improvement
+
+It is vital that the whole team – the Product Owner, Scrum Master, and Team Members, work together to align sustainability goals with sprint goals, making sustainability a shared responsibility.
+The team should review the sustainability metrics they create in the sprint demo and discuss them in the retrospective. This will create a sustainable feedback loop as part of your agile process, allowing the team to continuously refine their approach to greener practices.
+
+# Conclusion
+
+When it comes to sustainable software, here at Scott Logic we try to follow the mantra is “Map, Measure, Mitigate!”
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Typo: remove "is" before Map, Measure, Mitigate

File: _posts/2024-09-03-sustainable-agile.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+- *Map*: Map your carbon emissions so that you know where the carbon lives in your system and can start to understand what you can influence and how you can influence it.
+- *Measure*: Do your best to apply appropriate measurements to the difference components in your system. Remember that some kind of measurement is better than nothing, so when you can’t measure something directly, measure an appropriate proxy that will indicate if your carbon emissions have changed. And if you can’t do that, work out a robust method to estimate the carbon emissions of that part of the system.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Typo: "difference components" => "different components"

File: _posts/2024-09-03-sustainable-agile.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+- *Mitigate*: As a team, each of you has a role in reducing the carbon emissions of your products. As a Product Owner, help your team to make time to reduce carbon emissions. As a Scrum Master make the data available and review it with your team. And as a Team Member, whatever your specialisation, use your skills to optimise your products and make them more efficient.
```
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: Slightly clunky sentence, try this:

> As a Product Owner, allow your team the time to reduce carbon emissions

File: _posts/2024-09-03-sustainable-agile.md (hunk 5 above)
Comment (chriswilty) [inline _posts/2024-09-03-sustainable-agile.md]: You have optimise/optimising with an "s" mostly in this blog, but two occurrences with a "z". Choose one over the other!

**Review by chriswilty — COMMENTED** [review]
Excellent! Fluent and comprehensive read, though I must admit, quite overwhelming...

I feel like I need a cheatsheet or a checklist that I can work through, effectively a summary of this post in bullet form.

File: _posts/2024-09-03-sustainable-agile.md [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+At the project level, you just need to focus on the items you want in scope for your team. Using the Upstream Emissions category (U), and Category O of Operational Emissions, you can track the hardware that your team uses to determine the teams own carbon emissions while developing the software. These categories can also be used if you have on premises server hardware that you use for development, testing or in production environments.
+
+Category C is used for cloud environments, software as a service and other IT services that you do not host yourself. If you are unsure what category to put a system into, ask if you pay the electricity bill for that system, or if you pay a third party for access to that system. If you are paying the electricity bill, it is classed as a direct emission (category O). Otherwise, it is indirect (category C).
+
+Your systems also have an impact downstream: 
+- the minimum specification of your software drives hardware sales and therefore old hardware disposal. 
+- software performance and network traffic use the resources of your consumers and the infrastructure used to reach them. 
+
+These impacts are mapped to Category D.
```
Comment (dogle-scottlogic) [inline _posts/2024-09-03-sustainable-agile.md]: Not sure this should be it's own paragraph? Seems to be part of the previous one?

File: _posts/2024-09-03-sustainable-agile.md [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+
+These are just a few of the tools available right now and there are more appearing every day.
+
+# How Scrum Roles Can Mitigate Carbon Emissions
+
+Having talked about how we map and measure the carbon emissions of the project, it is down to the team to start reducing carbon emissions. Each role has their part to play.
+
+## If you are the Product Owner
+
+As a product owner, you are already responsible for defining the product’s features, liaising with stakeholders, and prioritising the backlog.
```
Comment (dogle-scottlogic) [inline _posts/2024-09-03-sustainable-agile.md]: Capitals for Product Owner?

File: _posts/2024-09-03-sustainable-agile.md (hunk 5 above)
Comment (dogle-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: I thought the same thing, maybe more realistic to say something more like "be mindful of features that reduce the products environmental footprint...when prioritising features"?

File: _posts/2024-09-03-sustainable-agile.md [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+
+- *Feature Prioritization*: Try to prioritize features that reduce the product's environmental footprint, such as optimizing code efficiency or reducing server load. 
+- *Estimation*: Encourage team members to incorporate sustainable considerations into their estimates. Help them to carve out time to craft high quality software – remember, doing so can help reduce your running costs and total cost of ownership by keeping systems well designed and efficient.
```
Comment (dogle-scottlogic) [inline _posts/2024-09-03-sustainable-agile.md]: is estimation part of the POs role? I might fit better under Scrum Master or Team Member?

File: _posts/2024-09-03-sustainable-agile.md (hunk 3 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: Fixed

File: _posts/2024-09-03-sustainable-agile.md (hunk 4 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: Agreed - added a clarification.

File: _posts/2024-09-03-sustainable-agile.md [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+
+You can integrate the sustainability metrics discussed above into decision-making, ensuring that carbon reduction is a key factor in planning and prioritization. For example, if you had a user story that would allow a product to use a smaller, or even eliminate a cloud appliance, prioritising it would reflect in the cloud emissions metrics.
+
+## If you are the Scrum Master
+
+As the Scrum Master you ensure the team adheres to Agile principles and facilitates processes.
+
+- *Process Optimization*: You can identify and eliminate wasteful practices in the development process, such as unnecessary meetings or redundant testing, that consume excessive energy.
+- *Green Agile Practices*: Promote the use of energy-efficient tools and methods and ensure that sustainability is part of the team’s definition of done.
+- *Integrating Technology Carbon Standard Practices*: The Scrum Master can educate the team on the Technology Carbon Standard, embedding it into the team's continuous improvement practices.
```
Comment (dogle-scottlogic) [inline _posts/2024-09-03-sustainable-agile.md]: Do you mean things like making it part of retro? Might be worth calling that out if so

File: _posts/2024-09-03-sustainable-agile.md (hunk 9 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: This kind of comes from the above conversation on measurement. I've expanded this a little to make that connection clearer. Does this work better now?

File: _posts/2024-09-03-sustainable-agile.md (hunk 5 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: Good point. I've added a little commentary to explain why it is probably a good idea above and beyond being "green".

**Review by dogle-scottlogic — COMMENTED** [review]
Great read - feels like maybe there's something more at the end could be added to tie together how agile can be used for good in this context if we're careful. You mentioned agile being about small increments but it's also about bringing technical and business people together and an approach of - do a small thing, check where you are, then adjust - which can align nicely

File: _posts/2024-09-03-sustainable-agile.md (hunk 5 above)
Comment (chriswilty) [reply _posts/2024-09-03-sustainable-agile.md]: Lovely 👌

File: _posts/2024-09-03-sustainable-agile.md (hunk 9 above)
Comment (chriswilty) [reply _posts/2024-09-03-sustainable-agile.md]: Yes that looks good.

File: _posts/2024-09-03-sustainable-agile.md (hunk 1 above)
Comment (chriswilty) [reply _posts/2024-09-03-sustainable-agile.md]: Yes, worth tagging despite currently tags being unused: I am about to add an issue to add tags as keywords, which should be applied retroactively if accepted. Add as many tags as you wish 😉

File: _posts/2024-09-03-sustainable-agile.md (hunk 15 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: I'll re-word this a little and add a point under the other two. I'm trying to advise POs to expect and encourage estimates to include sustainability as a consideration, as I would expect them to do so for other NFRs like security and performance.

File: _posts/2024-09-03-sustainable-agile.md (hunk 16 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: I'm planning a follow up post at some point to cover specific ceremonies. This post is big enough as it is...

File: _posts/2024-09-03-sustainable-agile.md (hunk 13 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: Reworded to combine into one paragraph.

File: _posts/2024-09-03-sustainable-agile.md (hunk 14 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md]: Fixed

File: _posts/2024-09-03-sustainable-agile.md (hunk 16 above)
Comment (jcamilleri-scottlogic) [reply _posts/2024-09-03-sustainable-agile.md:107]: Updated

**Review by dogle-scottlogic — APPROVED** [review]

**Pull request merged**